runtimevar/awssecretsmanager/awssecretsmanager.go

```diff
@@ -0,0 +1,359 @@
+// See gocloud.dev/aws/ConfigFromURLParams for supported query parameters
+// that affect the default AWS session.
+//
+// In addition, the following URL parameters are supported:
+//   - decoder: The decoder to use. Defaults to URLOpener.Decoder, or
+//       runtimevar.BytesDecoder if URLOpener.Decoder is nil.
+//       See runtimevar.DecoderByName for supported values.
+type URLOpener struct {
+	// ConfigProvider must be set to a non-nil value.
+	ConfigProvider client.ConfigProvider
+
+	// Decoder specifies the decoder to use if one is not specified in the URL.
+	// Defaults to runtimevar.BytesDecoder.
+	Decoder *runtimevar.Decoder
+
+	// Options specifies the options to pass to New.
+	Options Options
+}
+
+// lazySessionOpener obtains the AWS session from the environment on the first
+// call to OpenVariableURL.
+type lazySessionOpener struct {
+	init   sync.Once
+	opener *URLOpener
+	err    error
+}
+
+func (o *lazySessionOpener) OpenVariableURL(ctx context.Context, u *url.URL) (*runtimevar.Variable, error) {
+	o.init.Do(func() {
+		sess, err := gcaws.NewDefaultSession()
+		if err != nil {
+			o.err = err
+			return
+		}
+		o.opener = &URLOpener{
+			ConfigProvider: sess,
+		}
+	})
+	if o.err != nil {
+		return nil, fmt.Errorf("open variable %v: %v", u, o.err)
+	}
+	return o.opener.OpenVariableURL(ctx, u)
+}
+
+// Scheme is the URL scheme awssecretsmanager registers its URLOpener under on runtimevar.DefaultMux.
+const Scheme = "awssecretsmanager"
+
+// OpenVariableURL opens the variable at the URL's path. See the package doc
+// for more details.
+func (o *URLOpener) OpenVariableURL(ctx context.Context, u *url.URL) (*runtimevar.Variable, error) {
+	q := u.Query()
+
+	decoderName := q.Get("decoder")
+	q.Del("decoder")
+	decoder, err := runtimevar.DecoderByName(ctx, decoderName, o.Decoder)
+	if err != nil {
+		return nil, fmt.Errorf("open variable %v: invalid decoder: %v", u, err)
+	}
+
+	configProvider := &gcaws.ConfigOverrider{
+		Base: o.ConfigProvider,
+	}
+	overrideCfg, err := gcaws.ConfigFromURLParams(q)
+	if err != nil {
+		return nil, fmt.Errorf("open variable %v: %v", u, err)
+	}
+
+	configProvider.Configs = append(configProvider.Configs, overrideCfg)
+
+	return OpenVariable(configProvider, path.Join(u.Host, u.Path), decoder, &o.Options)
+}
+
+// Options sets options.
+type Options struct {
+	// WaitDuration controls the rate at which AWS Secrets Manager is polled.
+	// Defaults to 30 seconds.
+	WaitDuration time.Duration
+}
+
+// OpenVariable constructs a *runtimevar.Variable backed by the variable secretID in AWS Secrets Manager.
+// Secrets Manager returns raw bytes; provide a decoder to decode the raw bytes
+// into the appropriate type for runtimevar.Snapshot.Value.
+// See the runtimevar package documentation for examples of decoders.
+func OpenVariable(sess client.ConfigProvider, name string, decoder *runtimevar.Decoder, opts *Options) (*runtimevar.Variable, error) {
```

Comment:
name -> secretID

Reply:
AWS Secrets Manager doc states:

> 	// Specifies the secret containing the version that you want to retrieve. You
	// can specify either the Amazon Resource Name (ARN) or the friendly name of
	// the secret.

Thus, I think it would be better to keep it as `name` for better clarity and consistency with other drivers.